src/PostgREST/AppState.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module PostgREST.AppState
+  ( AppState
+  , init
+  , initWithPool
+  , pool
+  , releasePool
+  , pgVersion
+  , isWorkerOn
+  , time
+  , config
+  , dbStructure
+  , putDbStructure
+  , putIsWorkerOn
+  , putPgVersion
+  , putConfig
+  , killMainThread
+  ) where
+
+import qualified Hasql.Pool as P
+
+import Control.AutoUpdate (defaultUpdateSettings, mkAutoUpdate,
+                           updateAction)
+import Data.IORef         (IORef, atomicWriteIORef, newIORef,
+                           readIORef)
+import Data.Time.Clock    (UTCTime, getCurrentTime)
+
+import PostgREST.Config                (AppConfig (..))
+import PostgREST.DbStructure           (DbStructure)
+import PostgREST.DbStructure.PgVersion (PgVersion (..))
+
+import Protolude      hiding (toS)
+import Protolude.Conv (toS)
+
+
+data AppState = AppState
+  { statePool         :: P.Pool
+  -- ^ Connection pool, either a 'Connection' or a 'ConnectionError'
+  , statePgVersion    :: MVar PgVersion
+  -- ^ Used to sync the listener(NOTIFY reload) with the connectionWorker. No
+  -- connection for the listener at first. Only used if dbChannelEnabled=true.
+  , stateDbStructure  :: IORef (Maybe DbStructure)
+  -- ^ No schema cache at the start. Will be filled in by the connectionWorker
+  , stateIsWorkerOn   :: IORef Bool
+  -- ^ Helper ref to make sure just one connectionWorker can run at a time
+  , stateConf         :: IORef AppConfig
+  -- ^ Config that can change at runtime
+  , stateGetTime      :: IO UTCTime
+  , stateMainThreadId :: ThreadId
+  }
```

Comment:
Weird, to have the comments **below** the fields. Everywhere else we put them above.